growth/too/tasks/ztf_client.py

```diff
@@ -89,3 +93,68 @@ def ztf_references():
                          reference_filter_ids=rows['fid'].tolist(),
                          reference_filter_mags=rows['maglimit'].tolist()))
     models.db.session.commit()
+
+
+@celery.task(base=PeriodicTask, shared=False, run_every=3600)
+def ztf_depot(start_time=None, end_time=None):
+    """ 
+    ZTF depot reader, ingesting information about images from
+    all program ids (including program_id = 1). 
+  
+    Parameters: 
+        start_time (astropy.Time): start time of request. 
+        end_time (astropy.Time): end time of request.    
+    """
+
+    if start_time is None:
+        start_time = time.Time.now() - time.TimeDelta(1.0*u.day)
+    if end_time is None:
+        end_time = time.Time.now()
+
+    depotdir = 'https://ztfweb.ipac.caltech.edu/ztf/depot'
+
+    mjds = np.arange(np.floor(start_time.mjd), np.ceil(end_time.mjd))
+    for mjd in mjds:
+        this_time = time.Time(mjd, format='mjd')
+        dstr = this_time.iso.split(" ")[0].replace("-", "")
+
+        filename = os.path.join(depotdir, '%s/goodsubs_%s.txt' % (dstr, dstr))
+        r = requests.get(filename)
+        lines = r.text.split("\n")
+        names = ['jd', 'field', 'rcid', 'ra0', 'dec0',
+                 'nalertpackets', 'programid', 'expid', 'fid',
+                 'scimaglim', 'diffmaglim', 'sciinpseeing', 'difffwhm']
+        data = Table.read(r.text, format='ascii.fixed_width',
+                   data_start=2, data_end=-1)
+        if len(data) == 0:
+            continue
+        obstable = Table(names=names, data=data)
+
+        obs_grouped_by_jd = obstable.group_by('jd').groups
+        for jd, rows in zip(obs_grouped_by_jd.keys, obs_grouped_by_jd):
+            for row in rows:
+                obstime = time.Time(row['jd'], format='jd').datetime,
+                models.db.session.merge(
+                    models.Observation(telescope='ZTF',
+                                       field_id=int(row['field']),
```

Comment:
Why are these casts necessary?

Reply:
https://stackoverflow.com/questions/50626058/psycopg2-cant-adapt-type-numpy-int64?rq=1

The numpy.int64 to int thing.

Reply:
But that only applies to arrays, right? These are scalars.